drivers/media/i2c/ov2311.c

```diff
@@ -0,0 +1,1205 @@
+	int ret;
+
+	if (ov2311->streaming) {
+		ret = ov2311_start_stream(ov2311);
+		if (ret)
+			goto error;
+	}
+	return 0;
+
+error:
+	ov2311_stop_stream(ov2311);
+	ov2311->streaming = 0;
+	return ret;
+}
+
+static int ov2311_runtime_suspend(struct device *dev)
+{
+	struct v4l2_subdev *sd = dev_get_drvdata(dev);
+	struct ov2311 *ov2311 = to_ov2311(sd);
+
+	if (ov2311->streaming)
+		ov2311_stop_stream(ov2311);
+
+	return 0;
+}
+
+static int ov2311_open(struct v4l2_subdev *sd, struct v4l2_subdev_fh *fh)
+{
+	struct ov2311 *ov2311 = to_ov2311(sd);
+	struct v4l2_mbus_framefmt *try_fmt =
+				v4l2_subdev_get_try_format(sd, fh->state, 0);
+	const struct ov2311_mode *def_mode = &supported_modes[0];
+
+	mutex_lock(&ov2311->mutex);
+	/* Initialize try_fmt */
+	try_fmt->width = def_mode->width;
+	try_fmt->height = def_mode->height;
+	try_fmt->code = MEDIA_BUS_FMT_Y10_1X10;
+	try_fmt->field = V4L2_FIELD_NONE;
+	try_fmt->colorspace = V4L2_COLORSPACE_RAW;
+	try_fmt->ycbcr_enc = V4L2_MAP_YCBCR_ENC_DEFAULT(try_fmt->colorspace);
+	try_fmt->quantization =
+		V4L2_MAP_QUANTIZATION_DEFAULT(true, try_fmt->colorspace,
+					      try_fmt->ycbcr_enc);
+	try_fmt->xfer_func = V4L2_MAP_XFER_FUNC_DEFAULT(try_fmt->colorspace);
+
+	mutex_unlock(&ov2311->mutex);
+	/* No crop or compose */
+
+	return 0;
+}
+
+static const struct dev_pm_ops ov2311_pm_ops = {
+	SET_RUNTIME_PM_OPS(ov2311_runtime_suspend,
+			   ov2311_runtime_resume, NULL)
+	SET_RUNTIME_PM_OPS(ov2311_power_off, ov2311_power_on, NULL)
+};
+
+static const struct v4l2_subdev_internal_ops ov2311_internal_ops = {
+	.open = ov2311_open,
+};
+
+static const struct v4l2_subdev_core_ops ov2311_core_ops = {
+	.subscribe_event = v4l2_ctrl_subdev_subscribe_event,
+	.unsubscribe_event = v4l2_event_subdev_unsubscribe,
+};
+
+static const struct v4l2_subdev_video_ops ov2311_video_ops = {
+	.s_stream = ov2311_s_stream,
+};
+
+static const struct v4l2_subdev_pad_ops ov2311_pad_ops = {
+	.enum_mbus_code = ov2311_enum_mbus_code,
+	.enum_frame_size = ov2311_enum_frame_sizes,
+	.get_fmt = ov2311_get_fmt,
+	.set_fmt = ov2311_set_fmt,
+	.get_selection = ov2311_get_selection,
+};
+
+static const struct v4l2_subdev_ops ov2311_subdev_ops = {
+	.core	= &ov2311_core_ops,
+	.video	= &ov2311_video_ops,
+	.pad	= &ov2311_pad_ops,
+};
+
+static int ov2311_set_ctrl(struct v4l2_ctrl *ctrl)
+{
+	struct ov2311 *ov2311 = container_of(ctrl->handler,
+					     struct ov2311, ctrl_handler);
+	struct i2c_client *client = ov2311->client;
+	s64 max;
+	int ret = 0;
+
+	/* Propagate change of current control to all related controls */
+	switch (ctrl->id) {
+	case V4L2_CID_VBLANK:
+		/* Update max exposure while meeting expected vblanking */
+		max = ov2311->cur_mode->height + ctrl->val - 4;
+		__v4l2_ctrl_modify_range(ov2311->exposure,
+					 ov2311->exposure->minimum, max,
+					 ov2311->exposure->step,
+					 ov2311->exposure->default_value);
+		break;
+	}
+
+	if (pm_runtime_get(&client->dev) <= 0)
+		return 0;
+	switch (ctrl->id) {
```

Comment:
The rest of the code is so nicely formatted that I couldn't help noticing the missing blank line before the switch compared to the one after. Either would be OK, it's the asymmetry that's noticeable.

Reply:
Blank line added.